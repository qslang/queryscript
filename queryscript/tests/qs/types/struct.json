{"a": 1, "b": {"c": 3}}
{"a": 1, "b": {"d": "cat"}}
