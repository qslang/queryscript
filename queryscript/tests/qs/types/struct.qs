let structs = load('struct.json');
select * from structs;
select * from structs where b = {c: 3, d: null};
select * from structs where b = {"c": 3, "d": null};

let foo = 1;
{"a": foo};
{foo: 1};
