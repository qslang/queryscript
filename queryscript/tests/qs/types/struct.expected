{
    "compile_errors": [],
    "decls": {
        "let foo": Int64,
        "let structs": External<[{
        	a UInt64,
        	b {
        	c UInt64,
        	d Utf8,
        },
        }]>,
    },
    "queries": [
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "a",
                                type_: Atom(
                                    UInt64,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "b",
                                type_: Record(
                                    [
                                        Field {
                                            name: "c",
                                            type_: Atom(
                                                UInt64,
                                            ),
                                            nullable: true,
                                        },
                                        Field {
                                            name: "d",
                                            type_: Atom(
                                                Utf8,
                                            ),
                                            nullable: true,
                                        },
                                    ],
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| a | b                 |\n|---|-------------------|\n| 1 | {c: 3, d: null}   |\n| 1 | {c: null, d: cat} |",
                viz: None,
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "a",
                                type_: Atom(
                                    UInt64,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "b",
                                type_: Record(
                                    [
                                        Field {
                                            name: "c",
                                            type_: Atom(
                                                UInt64,
                                            ),
                                            nullable: true,
                                        },
                                        Field {
                                            name: "d",
                                            type_: Atom(
                                                Utf8,
                                            ),
                                            nullable: true,
                                        },
                                    ],
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| a | b               |\n|---|-----------------|\n| 1 | {c: 3, d: null} |",
                viz: None,
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "a",
                                type_: Atom(
                                    UInt64,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "b",
                                type_: Record(
                                    [
                                        Field {
                                            name: "c",
                                            type_: Atom(
                                                UInt64,
                                            ),
                                            nullable: true,
                                        },
                                        Field {
                                            name: "d",
                                            type_: Atom(
                                                Utf8,
                                            ),
                                            nullable: true,
                                        },
                                    ],
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| a | b               |\n|---|-----------------|\n| 1 | {c: 3, d: null} |",
                viz: None,
            },
        ),
        Ok(
            TypedValue {
                type_: Record(
                    [
                        Field {
                            name: "a",
                            type_: Atom(
                                Int64,
                            ),
                            nullable: true,
                        },
                    ],
                ),
                value: "{a: 1}",
                viz: None,
            },
        ),
        Ok(
            TypedValue {
                type_: Record(
                    [
                        Field {
                            name: "foo",
                            type_: Atom(
                                Int64,
                            ),
                            nullable: true,
                        },
                    ],
                ),
                value: "{foo: 1}",
                viz: None,
            },
        ),
        Ok(
            TypedValue {
                type_: Record(
                    [],
                ),
                value: "{}",
                viz: None,
            },
        ),
    ],
}